src/NuGet.Core/NuGet.DependencyResolver.Core/Remote/RemoteDependencyWalker.cs

```diff
@@ -498,34 +516,71 @@ private static GraphItem<RemoteResolveResult> CreateUnresolvedMatch(LibraryRange
             }
         }
 
-        public Task<RemoteMatch> FindProjectMatch(string name, NuGetFramework framework, CancellationToken cancellationToken)
+        public Task<RemoteMatch> FindProjectMatch(
+            LibraryRange libraryRange,
+            NuGetFramework framework,
+            GraphEdge<RemoteResolveResult> outerEdge,
+            CancellationToken cancellationToken)
         {
             RemoteMatch result = null;
 
-            var libraryRange = new LibraryRange
+            // Check if projects are allowed for this dependency
+            if ((libraryRange.TypeConstraint & (LibraryTypeFlag.Project | LibraryTypeFlag.ExternalProject))
+                != LibraryTypeFlag.None)
             {
-                Name = name
-            };
+                // Find the root directory of the parent project if one exists.
+                // This is used for resolving global json.
+                var parentProjectRoot = GetRootPathForParentProject(outerEdge);
 
-            foreach (var provider in _context.ProjectLibraryProviders)
-            {
-                var match = provider.GetLibrary(libraryRange, framework);
-                if (match != null)
+                foreach (var provider in _context.ProjectLibraryProviders)
                 {
-                    result = new LocalMatch
+                    if (provider.SupportsType(libraryRange.TypeConstraint))
                     {
-                        LocalLibrary = match,
-                        Library = match.Identity,
-                        LocalProvider = provider,
-                        Provider = new LocalDependencyProvider(provider),
-                        Path = match.Path,
-                    };
+                        var match = provider.GetLibrary(libraryRange, framework, parentProjectRoot);
+                        if (match != null)
+                        {
+                            result = new LocalMatch
+                            {
+                                LocalLibrary = match,
+                                Library = match.Identity,
+                                LocalProvider = provider,
+                                Provider = new LocalDependencyProvider(provider),
+                                Path = match.Path,
+                            };
+                        }
+                    }
                 }
             }
 
             return Task.FromResult<RemoteMatch>(result);
         }
 
+        /// <summary>
+        /// Returns root directory of the parent project.
```

Comment:
I'm a little confused by this logic. Is this implementing the recursive walk up for `global.json` files as well? I need to double check the DNX behavior, but I think the behavior was for each project to have as their search paths for other projects, the union of: Their immediate parent directory, all directories specified in the first `global.json` file above them.